code/object/set.php

```diff
@@ -137,6 +95,29 @@ public function merge(ObjectSet $set)
     }
 
     /**
+     * Filter the set using a callback
+     *
+     * If the callback returns FALSE the object will not be included in the resulting object set.
+     *
+     * @param   Callable $filter A callback that will handle the filtering
+     * @return  ObjectSet Returns an object subset
+     */
+    public function filter(/*Callable*/ $filter)
```

Comment:
@johanjanssens can we implement this the other way around? Instead of clearing first and then re-adding later, we can call remove() for the objects that return false.